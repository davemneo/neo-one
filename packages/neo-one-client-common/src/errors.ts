import { makeErrorWithCode } from '@neo-one/utils';
import { common, PrivateKey } from './common';
import { ContractParameter } from './types';

/* istanbul ignore next */
export const UnknownOpError = makeErrorWithCode('UNKNOWN_OP', (byteCode: string) => `Unknown op: ${byteCode}`);
export const InvalidContractParameterError = makeErrorWithCode(
  'INVALID_CONTRACT_PARAMETER',
  (parameter: ContractParameter, expected: ReadonlyArray<ContractParameter['type']>) =>
    `Expected one of ${JSON.stringify(expected)} ` + `ContractParameterTypes, found ${parameter.type}`,
);
export const Base58CheckError = makeErrorWithCode(
  'BASE_58_CHECK',
  (value: string) => `Base58 Check Decode Error on decoding: ${value}`,
);
export const InvalidAddressError = makeErrorWithCode(
  'INVALID_ADDRESS',
  (address: string) => `Invalid Address: ${address}`,
);
export const TooManyPublicKeysError = makeErrorWithCode(
  'TOO_MANY_PUBLIC_KEYS',
  (amount: number) => `Too many public keys. Found: ${amount}, Max: 1024`,
);
export const InvalidNumberOfKeysError = makeErrorWithCode(
  'INVALID_NUMBER_OF_KEYS',
  (m: number, amount: number) =>
    `invalid number of keys. Found: ${m} keys, must be between 1 and ${amount} (number of public keys).`,
);
/* istanbul ignore next */
export const InvalidPrivateKeyError = makeErrorWithCode(
  'INVALID_PRIVATE_KEY',
  (privateKey: PrivateKey) => `Invalid Private Key, found: ${common.privateKeyToString(privateKey)}`,
);
export const InvalidSignatureError = makeErrorWithCode(
  'INVALID_SIGNATURE',
  (length: number) => `Invalid Signature length. Found: ${length}, Max: 64`,
);
export const InvalidSysCallError = makeErrorWithCode(
  'INVALID_SYS_CALL_NAME',
  (value: string) => `Expected sys call name, found: ${value}`,
);
export const InvalidVMStateError = makeErrorWithCode(
  'INVALID_VM_STATE',
  (state: number) => `Invalid VM State: ${state}`,
);
/* istanbul ignore next */
export const InvalidParamError = makeErrorWithCode(
  'INVALID_PARAM',
  (paramType?: string) => `Invalid Param${paramType === undefined ? '.' : `: ${paramType}`}`,
);
export const InvalidAssetTypeJSONError = makeErrorWithCode(
  'INVALID_ASSET_TYPE_JSON',
  (type: string) => `Invalid AssetType: ${type}`,
);
export const InvalidAssetTypeError = makeErrorWithCode(
  'INVALID_ASSET_TYPE',
  (assetType: number) => `Expected asset type, found: ${assetType.toString(16)}`,
);
export const InvalidStorageFlagsJSONError = makeErrorWithCode(
  'INVALID_STORAGE_FLAGS_JSON',
  (type: string) => `Invalid StorageFlags: ${type}`,
);
export const InvalidStorageFlagsError = makeErrorWithCode(
  'INVALID_STORAGE_FLAGS',
  (storageFlags: number) => `Expected StorageFlags, found: ${storageFlags.toString(16)}`,
);
export const InvalidStateDescriptorTypeError = makeErrorWithCode(
  'INVALID_STATE_DESCRIPTOR_TYPE',
  (stateDescriptorType: number) => `Expected StateDescriptorType, found: ${stateDescriptorType.toString(16)}`,
);
export const InvalidContractParameterTypeJSONError = makeErrorWithCode(
  'INVALID_CONTRACT_PARAMETER_TYPE_JSON',
  (value: string) => `Invalid ContractParameterType: ${value}`,
);
export const InvalidContractParameterTypeError = makeErrorWithCode(
  'INVALID_CONTRACT_PARAMETER_TYPE',
  (contractParameterType: number) => `Expected contract parameter type, found: ${contractParameterType.toString(16)}`,
);
export const InvalidAttributeUsageError = makeErrorWithCode(
  'INVALID_ATTRIBUTE_USAGE',
  (transactionAttributeUsage: number) =>
    `Expected transaction attribute usage, found: ${transactionAttributeUsage.toString(16)}`,
);
export const InvalidAttributeUsageJSONError = makeErrorWithCode(
  'INVALID_ATTRIBUTE_USAGE_JSON',
  (transactionAttributeUsage: string) => `Expected transaction attribute usage, found: ${transactionAttributeUsage}`,
);
export const InvalidTransactionTypeError = makeErrorWithCode(
  'INVALID_TRANSACTION_TYPE',
  (transactionType: number) => `Expected transaction type, found: ${transactionType.toString(16)}`,
);
export const InvalidVMByteCodeError = makeErrorWithCode(
  'INVALID_VM_OP_CODE',
  (value: number) => `Expected VM OpCode, received: ${value}}`,
);
export const InvalidSignaturesError = makeErrorWithCode(
  'INVALID_SIGNATURES',
  (m: number, value: number) => `Expected ${m} unique signatures, found: ${value}.`,
);
