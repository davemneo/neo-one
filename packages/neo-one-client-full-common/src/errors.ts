import { makeErrorWithCode } from '@neo-one/utils';

// tslint:disable-next-line export-name
export const InvalidContractPropertyStateError = makeErrorWithCode(
  'INVALID_CONTRACT_PROPERTY_STATE',
  (contractParameterType: number) => `Expected contract parameter type, found: ${contractParameterType.toString(16)}`,
);
export const InvalidVersionError = makeErrorWithCode(
  'INVALID_VERSION',
  (expected: number, value: number) => `Expected version to be ${expected}, found: ${value}`,
);
