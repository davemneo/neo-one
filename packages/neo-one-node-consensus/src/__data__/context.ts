import { common } from '@neo-one/client-common';
import _ from 'lodash';
import { cloneBlockSent, cloneInitial, Context } from '../context';
import { keys } from './keys';

const validators = keys.map(({ publicKey }) => publicKey);
const previousHash = common.stringToUInt256('0xd42561e3d30e15be6400b6df2f328e02d2bf6354c41dce433bc57687c82144bf');

const expectedView = _.range(0, 7).map(() => 0);
const blockReceivedTimeSeconds = 1513018010;

const viewNumber = 0;
const primaryIndex = 1;
const blockIndex = primaryIndex;
const backupIndex = 2;
const backupContext = new Context({
  type: 'backup',
  previousHash,
  blockIndex,
  viewNumber,
  myIndex: backupIndex,
  primaryIndex,
  expectedView,
  validators,
  blockReceivedTimeSeconds,
});

const backupPrivateKey = keys[backupIndex].privateKey;

const primaryContext = new Context({
  type: 'primary',
  previousHash,
  blockIndex,
  viewNumber,
  myIndex: primaryIndex,
  primaryIndex,
  expectedView,
  validators,
  blockReceivedTimeSeconds,
});

const primaryPrivateKey = keys[primaryIndex].privateKey;

const blockSentBackupContext = cloneBlockSent(backupContext);
const initialBackupContext = cloneInitial(backupContext, {
  type: 'backup',
  viewNumber,
  primaryIndex,
});

const blockSentPrimaryContext = cloneBlockSent(primaryContext);
const initialPrimaryContext = cloneInitial(primaryContext, {
  type: 'primary',
  viewNumber,
  primaryIndex,
});

export const context = {
  backupPrivateKey,
  primaryPrivateKey,
  previousHash,
  blockSentBackupContext,
  initialBackupContext,
  blockSentPrimaryContext,
  initialPrimaryContext,
};
