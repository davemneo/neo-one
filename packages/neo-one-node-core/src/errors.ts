import { makeErrorWithCode } from '@neo-one/utils';

// tslint:disable-next-line export-name
export const ReceiveMessageTimeoutError = makeErrorWithCode(
  'RECEIVE_MESSAGE_TIMEOUT',
  (endpoint: string) => `Message with endpoint ${endpoint} timed out.`,
);
export const InvalidActionTypeError = makeErrorWithCode(
  'INVALID_ACTION_TYPE',
  (type: number) => `Expected action type, found: ${type}`,
);
export const InvalidScriptContainerTypeError = makeErrorWithCode(
  'INVALID_SCRIPT_CONTAINER_TYPE',
  (value: number) => `Expected ScriptContainerType, found: ${value.toString(16)}`,
);
export const VerifyError = makeErrorWithCode(
  'VERIFY',
  (reason?: string) => `Verification failed${reason === undefined ? '.' : `: ${reason}`}`,
);
export const InvalidConsensusMessageTypeError = makeErrorWithCode(
  'INVALID_CONSENSUS_MESSAGE_TYPE',
  (value: number) => `Expected ConsensusMessageType, found: ${value.toString(16)}`,
);
export const UnsignedBlockError = makeErrorWithCode(
  'UNSIGNED_BLOCK',
  (stringHash: string) => `Block script does not exist because it has not been signed. @ block with hash ${stringHash}`,
);
