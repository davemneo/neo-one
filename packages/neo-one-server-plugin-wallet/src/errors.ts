import { makeErrorWithCode } from '@neo-one/utils';

const extraInfo = (extra?: string) => (extra === undefined ? '' : extra);

export const NetworkRequiredError = makeErrorWithCode('NETWORK_REQUIRED', () => 'Wallets require a network');
export const WalletRequiredError = makeErrorWithCode(
  'WALLET_REQUIRED',
  (extra?: string) => `Wallet required to perform operation.${extraInfo(extra)}`,
);
