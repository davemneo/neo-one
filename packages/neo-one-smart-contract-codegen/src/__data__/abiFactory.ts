import { ABI, ABIDefault, ABIReturn } from '@neo-one/client-common';

// tslint:disable-next-line:no-any
const getCombinations = (list: ReadonlyArray<ReadonlyArray<any>>) => {
  const floorsIn = list.reduceRight<ReadonlyArray<number>>((acc, right) => [right.length * acc[0]].concat(acc), [1]);
  const max = floorsIn[0];
  const floors = floorsIn.slice(1);

  const iteratorFuncs = list.map((item, index) =>
    index === list.length - 1
      ? (value: number) => item[value % item.length]
      : (value: number) => item[Math.floor(value / floors[index]) % item.length],
  );

  return [...Array(max)].map((_, index) => iteratorFuncs.map((func) => func(index)));
};

const abiFunctionOptions: ReadonlyArray<{ readonly [key in string]?: boolean }> = [
  { constant: true },
  { send: true },
  { sendUnsafe: true },
  { sendUnsafe: true, receive: true },
  { receive: true },
  { claim: true },
  { refundAssets: true },
  { completeSend: true },
  {},
];

const abiParameterOptions: ReadonlyArray<{ readonly [key in string]?: boolean | ABIDefault }> = [
  { default: { type: 'sender' } },
  { rest: true },
  { default: { type: 'sender' }, rest: true },
  {},
];

const abiReturns: ReadonlyArray<ABIReturn> = [
  {
    type: 'ForwardValue',
  },
  {
    type: 'Boolean',
    optional: true,
  },
  {
    type: 'Boolean',
    forwardedValue: true,
  },
  {
    type: 'Integer',
    decimals: 4,
  },
  {
    type: 'Array',
    value: {
      type: 'Boolean',
    },
  },
  {
    type: 'Map',
    key: { type: 'Integer', decimals: 0 },
    value: { type: 'Boolean' },
  },
  {
    type: 'Object',
    properties: {
      property: { type: 'Boolean' },
    },
  },
];

const boolReturnType: ABIReturn = { type: 'Boolean' };

const createBaseABIFunctions = (): ReadonlyArray<ABI> => {
  const combinations = getCombinations([abiReturns, abiFunctionOptions]);

  return combinations.map(([abiReturn, abiFuncProps]) => {
    const funcPropNames = Object.keys(abiFuncProps);
    const name = `${abiReturn.type}${`${funcPropNames.length === 0 ? '' : '_'}${funcPropNames.join('_')}`}`;

    return {
      functions: [
        {
          name,
          returnType: abiReturn,
          ...abiFuncProps,
        },
      ],
    };
  });
};

const createBaseABIEvents = (): ReadonlyArray<ABI> => {
  const combinations = getCombinations([abiReturns, abiParameterOptions]);

  return combinations.map(([abiReturn, abiParamProps]) => {
    const paramPropNames = Object.keys(abiParamProps);
    const name = `${abiReturn.type}${`${paramPropNames.length === 0 ? '' : '_'}${paramPropNames.join('_')}`}`;

    return {
      functions: [
        {
          name: 'foo',
          returnType: boolReturnType,
        },
      ],
      events: [
        {
          name,
          parameters: [
            {
              name: 'arg',
              ...abiReturn,
              ...abiParamProps,
            },
          ],
        },
      ],
    };
  });
};

const createBaseABIFunctionParameters = (): ReadonlyArray<ABI> => {
  const combinations = getCombinations([abiReturns, abiParameterOptions]);

  return combinations.map(([abiReturn, abiParamProps]) => {
    const paramPropNames = Object.keys(abiParamProps);
    const name = `${abiReturn.type}${`${paramPropNames.length === 0 ? '' : '_'}${paramPropNames.join('_')}`}`;

    return {
      functions: [
        {
          name,
          parameters: [
            {
              name: 'arg',
              ...abiReturn,
              ...abiParamProps,
            },
          ],
          returnType: boolReturnType,
        },
      ],
    };
  });
};

export const abiFactory = {
  createBaseABIFunctions,
  createBaseABIFunctionParameters,
  createBaseABIEvents,
};
