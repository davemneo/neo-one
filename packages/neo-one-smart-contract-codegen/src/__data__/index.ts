export * from './nep5';
export * from './testUtils';
export * from './abiFactory';
