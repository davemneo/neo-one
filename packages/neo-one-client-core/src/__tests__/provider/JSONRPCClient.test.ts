// tslint:disable no-object-mutation
import { data, factory, keys } from '../../__data__';
import { JSONRPCError, RelayTransactionError } from '../../errors';
import { JSONRPCClient, JSONRPCProvider } from '../../provider';

const mockRequest = jest.fn();
class MockJSONRPCProvider extends JSONRPCProvider {
  // tslint:disable-next-line readonly-keyword
  public request = mockRequest;
}

describe('JSONRPCClient', () => {
  const provider = new MockJSONRPCProvider();
  const client = new JSONRPCClient(provider);

  beforeEach(() => {
    mockRequest.mockReset();
  });

  test('getAccount', async () => {
    const value = factory.createAccountJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getAccount(keys[0].address);

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getAsset', async () => {
    const value = factory.createAssetJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getAsset(data.hash256s.a);

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getBlock - hash', async () => {
    const value = factory.createBlockJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getBlock(data.hash256s.a);

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getBlock - index', async () => {
    const value = factory.createBlockJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getBlock(10, { timeoutMS: 1000 });

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getBestBlockHash', async () => {
    const value = data.hash256s.a;
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getBestBlockHash();

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getBlockCount', async () => {
    const value = 10;
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getBlockCount();

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getContract', async () => {
    const value = factory.createContractJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getContract(keys[0].address);

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getMemPool', async () => {
    const value = [data.hash256s.a];
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getMemPool();

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getTransaction', async () => {
    const value = factory.createInvocationTransactionJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getTransaction(data.hash256s.a);

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getUnspentOutput - defined', async () => {
    const value = factory.createOutputJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getUnspentOutput(factory.createInputJSON());

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getUnspentOutput - undefined', async () => {
    const value = undefined;
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getUnspentOutput(factory.createInputJSON());

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('testInvokeRaw', async () => {
    const value = factory.createInvocationResultSuccessJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.testInvokeRaw(data.buffers.a);

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('relayTransaction', async () => {
    const value = factory.createInvocationTransactionJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.relayTransaction(data.buffers.a);

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('relayTransaction - relay error', async () => {
    const message = 'error';
    const rpcErr = new JSONRPCError({ message, code: -110 });
    mockRequest.mockImplementationOnce(async () => Promise.reject(rpcErr));

    const result = client.relayTransaction(data.buffers.a);

    await expect(result).rejects.toEqual(new RelayTransactionError(message));
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('relayTransaction - other error', async () => {
    const error = new Error('error');
    mockRequest.mockImplementationOnce(async () => Promise.reject(error));

    const result = client.relayTransaction(data.buffers.a);

    await expect(result).rejects.toEqual(error);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getOutput', async () => {
    const value = factory.createOutputJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getOutput(factory.createInputJSON());

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getClaimAmount', async () => {
    const value = data.bigNumbers.a.toString(10);
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getClaimAmount(factory.createInputJSON());

    expect(result.toString(10)).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('testInvocation', async () => {
    const value = factory.createCallReceiptJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.testInvocation(data.buffers.a);

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getTransactionReceipt', async () => {
    const value = factory.createTransactionReceipt();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getTransactionReceipt(data.hash256s.a, { timeoutMS: 1000 });

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getInvocationData', async () => {
    const value = factory.createInvocationDataJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getInvocationData(data.hash256s.a);

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getConnectedPeers', async () => {
    const value = { connected: [factory.createPeerJSON()] };
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getConnectedPeers();

    expect(result).toEqual(value.connected);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('getNetworkSettings', async () => {
    const value = factory.createNetworkSettingsJSON();
    mockRequest.mockImplementationOnce(async () => Promise.resolve(value));

    const result = await client.getNetworkSettings();

    expect(result).toEqual(value);
    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('runConsensusNow', async () => {
    mockRequest.mockImplementationOnce(async () => Promise.resolve());

    await client.runConsensusNow();

    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('updateSettings', async () => {
    mockRequest.mockImplementationOnce(async () => Promise.resolve());

    await client.updateSettings({ secondsPerBlock: 10 });

    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('fastForwardOffset', async () => {
    mockRequest.mockImplementationOnce(async () => Promise.resolve());

    await client.fastForwardOffset(10);

    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('fastForwardToTime', async () => {
    mockRequest.mockImplementationOnce(async () => Promise.resolve());

    await client.fastForwardToTime(10);

    expect(mockRequest.mock.calls).toMatchSnapshot();
  });

  test('reset', async () => {
    mockRequest.mockImplementationOnce(async () => Promise.resolve());

    await client.reset();

    expect(mockRequest.mock.calls).toMatchSnapshot();
  });
});
