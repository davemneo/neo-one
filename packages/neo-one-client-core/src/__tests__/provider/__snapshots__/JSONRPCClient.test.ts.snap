// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSONRPCClient fastForwardOffset 1`] = `
Array [
  Array [
    Object {
      "method": "fastforwardoffset",
      "params": Array [
        10,
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient fastForwardToTime 1`] = `
Array [
  Array [
    Object {
      "method": "fastforwardtotime",
      "params": Array [
        10,
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getAccount 1`] = `
Array [
  Array [
    Object {
      "method": "getaccountstate",
      "params": Array [
        "ALq7AWrhAueN6mJNqk6FHJjnsEoPRytLdW",
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getAllStorage 1`] = `
Array [
  Array [
    Object {
      "method": "getallstorage",
      "params": Array [
        "0xcef0c0fdcfe7838eff6ff104f9cdec2922297537",
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getAsset 1`] = `
Array [
  Array [
    Object {
      "method": "getassetstate",
      "params": Array [
        "0xb50034d97ba8c836758de1124b6c77d38bc772abc9a8f22c85e7389014e75234",
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getBestBlockHash 1`] = `
Array [
  Array [
    Object {
      "method": "getbestblockhash",
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getBlock - hash 1`] = `
Array [
  Array [
    Object {
      "method": "getblock",
      "params": Array [
        "0xb50034d97ba8c836758de1124b6c77d38bc772abc9a8f22c85e7389014e75234",
        1,
      ],
      "watchTimeoutMS": undefined,
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getBlock - index 1`] = `
Array [
  Array [
    Object {
      "method": "getblock",
      "params": Array [
        10,
        1,
      ],
      "watchTimeoutMS": 1000,
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getBlockCount 1`] = `
Array [
  Array [
    Object {
      "method": "getblockcount",
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getClaimAmount 1`] = `
Array [
  Array [
    Object {
      "method": "getclaimamount",
      "params": Array [
        "0xb50034d97ba8c836758de1124b6c77d38bc772abc9a8f22c85e7389014e75234",
        2,
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getConnectedPeers 1`] = `
Array [
  Array [
    Object {
      "method": "getpeers",
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getContract 1`] = `
Array [
  Array [
    Object {
      "method": "getcontractstate",
      "params": Array [
        "0xcef0c0fdcfe7838eff6ff104f9cdec2922297537",
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getInvocationData 1`] = `
Array [
  Array [
    Object {
      "method": "getinvocationdata",
      "params": Array [
        "0xb50034d97ba8c836758de1124b6c77d38bc772abc9a8f22c85e7389014e75234",
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getMemPool 1`] = `
Array [
  Array [
    Object {
      "method": "getrawmempool",
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getNetworkSettings 1`] = `
Array [
  Array [
    Object {
      "method": "getnetworksettings",
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getOutput 1`] = `
Array [
  Array [
    Object {
      "method": "getoutput",
      "params": Array [
        "0xb50034d97ba8c836758de1124b6c77d38bc772abc9a8f22c85e7389014e75234",
        2,
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getTransaction 1`] = `
Array [
  Array [
    Object {
      "method": "getrawtransaction",
      "params": Array [
        "0xb50034d97ba8c836758de1124b6c77d38bc772abc9a8f22c85e7389014e75234",
        1,
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getTransactionReceipt 1`] = `
Array [
  Array [
    Object {
      "method": "gettransactionreceipt",
      "params": Array [
        "0xb50034d97ba8c836758de1124b6c77d38bc772abc9a8f22c85e7389014e75234",
      ],
      "watchTimeoutMS": 1000,
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getUnspentOutput - defined 1`] = `
Array [
  Array [
    Object {
      "method": "gettxout",
      "params": Array [
        "0xb50034d97ba8c836758de1124b6c77d38bc772abc9a8f22c85e7389014e75234",
        2,
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient getUnspentOutput - undefined 1`] = `
Array [
  Array [
    Object {
      "method": "gettxout",
      "params": Array [
        "0xb50034d97ba8c836758de1124b6c77d38bc772abc9a8f22c85e7389014e75234",
        2,
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient relayTransaction - other error 1`] = `
Array [
  Array [
    Object {
      "method": "relaytransaction",
      "params": Array [
        "b500",
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient relayTransaction - relay error 1`] = `
Array [
  Array [
    Object {
      "method": "relaytransaction",
      "params": Array [
        "b500",
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient relayTransaction 1`] = `
Array [
  Array [
    Object {
      "method": "relaytransaction",
      "params": Array [
        "b500",
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient reset 1`] = `
Array [
  Array [
    Object {
      "method": "reset",
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient runConsensusNow 1`] = `
Array [
  Array [
    Object {
      "method": "runconsensusnow",
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient testInvocation 1`] = `
Array [
  Array [
    Object {
      "method": "testinvocation",
      "params": Array [
        "b500",
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient testInvokeRaw 1`] = `
Array [
  Array [
    Object {
      "method": "invokescript",
      "params": Array [
        "b500",
      ],
    },
    undefined,
  ],
]
`;

exports[`JSONRPCClient updateSettings 1`] = `
Array [
  Array [
    Object {
      "method": "updatesettings",
      "params": Array [
        Object {
          "secondsPerBlock": 10,
        },
      ],
    },
    undefined,
  ],
]
`;
